### PR TITLE
Upgrade ginkgo to v2

### PR DESCRIPTION
**Pull request by mpanchajanya**
### What this PR does / why we need it

- Upgrade ginkgo to ginkgo v2 v2.9.2
- Fixed E2E test on plugin compatibility
- To generate test results summary using ginkgo for coexistence tests and eventually for e2e tests we need this change.

### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR

make lint passed
make test passed
make e2e-cli-core-all passed

make tools - should generate ginkgo

`mpanchajanya@mpanchajanF09MK ~/v/r/tanzu-cli (update_ginkgo)> make tools
Screenshot 2023-04-19 at 13 41 01

`

### Release note

```release-note

```

### CODE REVIEW

**Review by chandrareddyp — APPROVED** [review]
LGTM!
Added a few nit's.

File: hack/tools/Makefile [hunk 1]
```diff
@@ -108,6 +110,11 @@ $(KIND):
 	mv ./kind $(@)
 	chmod a+x $(@)
 
+ginkgo: $(GINKGO) ## Install misspell
```
Comment (anujc25) [inline hack/tools/Makefile:113]:
```suggestion
ginkgo: $(GINKGO) ## Install ginkgo
```

**Comment by anujc25** [comment]
This is great @mpanchajanya. This also reduces the time of the `Test` step in the `Main / Build` pipeline from `12m` to `3m` which is significant.

**Review by anujc25 — APPROVED** [review]
LGTM. Just a minor comment.

**Pull request merged**